File: java/src/main/java/com/penapereira/patterns/command/Receiver.java
package com.penapereira.patterns.command;

import java.util.ArrayList;
import java.util.List;

/** Knows how to perform the operations associated with a request. */
public final class Receiver {

    private final List<String> words = new ArrayList<>();

    public void action(String word) {
        words.add(word);
    }

    public void reverse(String word) {
        int last = words.lastIndexOf(word);
        if (last >= 0) {
            words.remove(last);
        }
    }

    public String getState() {
        return String.join(" ", words);
    }
}

File: typescript/src/command/receiver.ts
/** Knows how to perform the operations associated with a request. */
export class Receiver {
  private readonly words: string[] = [];

  action(word: string): void {
    this.words.push(word);
  }

  reverse(word: string): void {
    const last = this.words.lastIndexOf(word);
    if (last >= 0) {
      this.words.splice(last, 1);
    }
  }

  getState(): string {
    return this.words.join(" ");
  }
}
